spring-cloud-sleuth-zipkin-stream/src/main/java/org/springframework/cloud/sleuth/zipkin/stream/ZipkinMessageListener.java

```diff
@@ -196,6 +191,13 @@ public Cloud cloud() {
 			return new CloudFactory().getCloud();
 		}
 
+		// TODO: what config shoud we use? zipkin uses zipkin.collector.sample-rate
+		@Bean
+		TraceIdSampler traceIdSampler() {
```

Comment:
So there's 2 tiers of sampling: instrumentation and collection

#90 is a number the app controls (affects instrumentation policy).We'd expect folks to have different values depending on throughput of their apps and probably its canary-ness.

This PR is a global rate, meaning that of all the spans sent from instrumentation, how many do we retain? This defaults to 100%, but you could lower this automatically or manually if, for example the storage system is degraded.

Make sense?